### PR TITLE
feat!: add ability to configure the cdn_policy on the managed compute_backend_service

### LINKED ISSUE #239: How do I configure the cdn_policy for backends?
I used the **serverless_negs** module to create backends using `enable_cdn = true`. Is there a way to configure the `cdn_policy` as well?

### PR DESCRIPTION
**Pull request by benhinchley**
This add support for configuring the `cdn_policy` for the `compute_backend_service` created by this module. 

fixes #239

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_terraform
 Failed Terraform check. More details below. 
 ```bash 
Running terraform fmt
Running terraform validate
terraform_validate . 
╷
│
 
Error: 
Unsupported argument
│
 
│
 
  on main.tf line 209, in resource "google_compute_backend_service" "default":
│
 
 209:       
deafult_ttl
 = lookup(lookup(each.value, "cdn_config", {}), "default_ttl", null)
│
 
│
 
An argument named "deafult_ttl" is not expected here. Did you mean
│
 
"default_ttl"?
╵
terraform_validate ./examples/cloudrun 
Success!
 The configuration is valid.
terraform_validate ./examples/https-gke 
╷
│
 
Warning: 
Argument is deprecated
│
 
│
 
  with tls_self_signed_cert.example,
│
 
  on tls.tf line 23, in resource "tls_self_signed_cert" "example":
│
 
  23:   key_algorithm   = 
tls_private_key.example.algorithm
│
 
│
 
This is now ignored, as the key algorithm is inferred from the
│
 
`private_key_pem`.
╵
╷
│
 
Error: 
Invalid value for module argument
│
 
│
 
  on main.tf line 49, in module "gce-lb-https":
│
 
  49:   backends = 
{
│
 
  50:     default = {
│
 
  51:       description                     = null
│
 
  52:       protocol                        = "HTTP"
│
 
  53:       port                            = var.service_port
│
 
  54:       port_name                       = var.service_port_name
│
 
  55:       timeout_sec                     = 10
│
 
  56:       connection_draining_timeout_sec = null
│
 
  57:       enable_cdn                      = false
│
 
  58:       security_policy                 = null
│
 
  59:       session_affinity                = null
│
 
  60:       affinity_cookie_ttl_sec         = null
│
 
  61:       custom_request_headers          = null
│
 
  62:       custom_response_headers         = null
│
 
  63:       health_check = {
│
 
  64:         check_interval_sec  = null
│
 
  65:         timeout_sec         = null
│
 
  66:         healthy_threshold   = null
│
 
  67:         unhealthy_threshold = null
│
 
  68:         request_path        = "/"
│
 
  69:         port                = var.service_port
│
 
  70:         host                = null
│
 
  71:         logging             = true
│
 
  72:       }
│
 
  73:       log_config = {
│
 
  74:         enable      = true
│
 
  75:         sample_rate = 1.0
│
 
  76:       }
│
 
  77:       groups = [
│
 
  78:         {
│
 
  79:           # Each node pool instance group should be added to the backend.
│
 
  80:           group                        = var.backend
│
 
  81:           balancing_mode               = null
│
 
  82:           capacity_scaler              = null
│
 
  83:           description                  = null
│
 
  84:           max_connections              = null
│
 
  85:           max_connections_per_instance = null
│
 
  86:           max_connections_per_endpoint = null
│
 
  87:           max_rate                     = null
│
 
  88:           max_rate_per_instance        = null
│
 
  89:           max_rate_per_endpoint        = null
│
 
  90:           max_utilization              = null
│
 
  91:         },
│
 
  92:       ]
│
 
  93:       iap_config = {
│
 
  94:         enable               = false
│
 
  95:         oauth2_client_id     = ""
│
 
  96:         oauth2_client_secret = ""
│
 
  97:       }
│
 
  98:     }
│
 
  99:   }
│
 
│
 
The given value is not suitable for child module variable "backends"
│
 
defined at ../../variables.tf:81,1-20: element "default": attribute
│
 
"cdn_config" is required.
╵
terraform_validate ./examples/https-gke/gke-node-port 
Success!
 The configuration is valid.
terraform_validate ./examples/https-redirect 
╷
│
 
Warning: 
Argument is deprecated
│
 
│
 
  with tls_self_signed_cert.example,
│
 
  on tls.tf line 23, in resource "tls_self_signed_cert" "example":
│
 
  23:   key_algorithm   = 
tls_private_key.example.algorithm
│
 
│
 
This is now ignored, as the key algorithm is inferred from the
│
 
`private_key_pem`.
╵
Success!
 The configuration is valid, but there were some
validation warnings as shown above.
terraform_validate ./examples/mig-nat-http-lb 
╷
│
 
Error: 
Invalid value for module argument
│
 
│
 
  on main.tf line 101, in module "gce-lb-http":
│
 
 101:   backends = 
{
│
 
 102:     default = {
│
 
 103:       description                     = null
│
 
 104:       protocol                        = "HTTP"
│
 
 105:       port                            = 80
│
 
 106:       port_name                       = "http"
│
 
 107:       timeout_sec                     = 10
│
 
 108:       connection_draining_timeout_sec = null
│
 
 109:       enable_cdn                      = false
│
 
 110:       security_policy                 = null
│
 
 111:       session_affinity                = null
│
 
 112:       affinity_cookie_ttl_sec         = null
│
 
 113:       custom_request_headers          = null
│
 
 114:       custom_response_headers         = null
│
 
 115:       health_check = {
│
 
 116:         check_interval_sec  = null
│
 
 117:         timeout_sec         = null
│
 
 118:         healthy_threshold   = null
│
 
 119:         unhealthy_threshold = null
│
 
 120:         request_path        = "/"
│
 
 121:         port                = 80
│
 
 122:         host                = null
│
 
 123:         logging             = null
│
 
 124:       }
│
 
 125:       log_config = {
│
 
 126:         enable      = false
│
 
 127:         sample_rate = null
│
 
 128:       }
│
 
 129:       groups = [
│
 
 130:         {
│
 
 131:           group                        = module.mig.instance_group
│
 
 132:           balancing_mode               = null
│
 
 133:           capacity_scaler              = null
│
 
 134:           description                  = null
│
 
 135:           max_connections              = null
│
 
 136:           max_connections_per_instance = null
│
 
 137:           max_connections_per_endpoint = null
│
 
 138:           max_rate                     = null
│
 
 139:           max_rate_per_instance        = null
│
 
 140:           max_rate_per_endpoint        = null
│
 
 141:           max_utilization              = null
│
 
 142:         }
│
 
 143:       ]
│
 
 144:       iap_config = {
│
 
 145:         enable               = false
│
 
 146:         oauth2_client_id     = ""
│
 
 147:         oauth2_client_secret = ""
│
 
 148:       }
│
 
 149:     }
│
 
 150:   }
│
 
│
 
The given value is not suitable for child module variable "backends"
│
 
defined at ../../variables.tf:81,1-20: element "default": attribute
│
 
"cdn_config" is required.
╵
terraform_validate ./examples/multi-backend-multi-mig-bucket-https-lb 
╷
│
 
Warning: 
Argument is deprecated
│
 
│
 
  with tls_self_signed_cert.example,
│
 
  on tls.tf line 23, in resource "tls_self_signed_cert" "example":
│
 
  23:   key_algorithm   = 
tls_private_key.example.algorithm
│
 
│
 
This is now ignored, as the key algorithm is inferred from the
│
 
`private_key_pem`.
╵
Success!
 The configuration is valid, but there were some
validation warnings as shown above.
terraform_validate ./examples/multi-mig-http-lb 
Success!
 The configuration is valid.
terraform_validate ./examples/multiple-certs 
╷
│
 
Warning: 
Argument is deprecated
│
 
│
 
  with tls_self_signed_cert.example,
│
 
  on tls.tf line 25, in resource "tls_self_signed_cert" "example":
│
 
  25:   key_algorithm   = 
tls_private_key.example[count.index].algorithm
│
 
│
 
This is now ignored, as the key algorithm is inferred from the
│
 
`private_key_pem`.
╵
╷
│
 
Error: 
Invalid value for module argument
│
 
│
 
  on main.tf line 139, in module "gce-lb-https":
│
 
 139:   backends = 
{
│
 
 140:     default = {
│
 
 141:       description                     = null
│
 
 142:       protocol                        = "HTTP"
│
 
 143:       port                            = 80
│
 
 144:       port_name                       = "http"
│
 
 145:       timeout_sec                     = 10
│
 
 146:       connection_draining_timeout_sec = null
│
 
 147:       enable_cdn                      = false
│
 
 148:       security_policy                 = null
│
 
 149:       session_affinity                = null
│
 
 150:       affinity_cookie_ttl_sec         = null
│
 
 151:       custom_request_headers          = null
│
 
 152:       custom_response_headers         = null
│
 
 153:       health_check = local.health_check
│
 
 154:       log_config = {
│
 
 155:         enable      = true
│
 
 156:         sample_rate = 1.0
│
 
 157:       }
│
 
 158:       groups = [
│
 
 159:         {
│
 
 160:           group                        = module.mig1.instance_group
│
 
 161:           balancing_mode               = null
│
 
 162:           capacity_scaler              = null
│
 
 163:           description                  = null
│
 
 164:           max_connections              = null
│
 
 165:           max_connections_per_instance = null
│
 
 166:           max_connections_per_endpoint = null
│
 
 167:           max_rate                     = null
│
 
 168:           max_rate_per_instance        = null
│
 
 169:           max_rate_per_endpoint        = null
│
 
 170:           max_utilization              = null
│
 
 171:         },
│
 
 172:         {
│
 
 173:           group                        = module.mig2.instance_group
│
 
 174:           balancing_mode               = null
│
 
 175:           capacity_scaler              = null
│
 
 176:           description                  = null
│
 
 177:           max_connections              = null
│
 
 178:           max_connections_per_instance = null
│
 
 179:           max_connections_per_endpoint = null
│
 
 180:           max_rate                     = null
│
 
 181:           max_rate_per_instance        = null
│
 
 182:           max_rate_per_endpoint        = null
│
 
 183:           max_utilization              = null
│
 
 184:         },
│
 
 185:         {
│
 
 186:           group                        = module.mig3.instance_group
│
 
 187:           balancing_mode               = null
│
 
 188:           capacity_scaler              = null
│
 
 189:           description                  = null
│
 
 190:           max_connections              = null
│
 
 191:           max_connections_per_instance = null
│
 
 192:           max_connections_per_endpoint = null
│
 
 193:           max_rate                     = null
│
 
 194:           max_rate_per_instance        = null
│
 
 195:           max_rate_per_endpoint        = null
│
 
 196:           max_utilization              = null
│
 
 197:         },
│
 
 198:       ]
│
 
 199:       iap_config = {
│
 
 200:         enable               = false
│
 
 201:         oauth2_client_id     = ""
│
 
 202:         oauth2_client_secret = ""
│
 
 203:       }
│
 
 204:     }
│
 
 205:     mig1 = {
│
 
 206:       description                     = null
│
 
 207:       protocol                        = "HTTP"
│
 
 208:       port                            = 80
│
 
 209:       port_name                       = "http"
│
 
 210:       timeout_sec                     = 10
│
 
 211:       connection_draining_timeout_sec = null
│
 
 212:       enable_cdn                      = false
│
 
 213:       security_policy                 = null
│
 
 214:       session_affinity                = null
│
 
 215:       affinity_cookie_ttl_sec         = null
│
 
 216:       custom_request_headers          = null
│
 
 217:       custom_response_headers         = null
│
 
 218:       health_check = local.health_check
│
 
 219:       log_config = {
│
 
 220:         enable      = true
│
 
 221:         sample_rate = 1.0
│
 
 222:       }
│
 
 223:       groups = [
│
 
 224:         {
│
 
 225:           group                        = module.mig1.instance_group
│
 
 226:           balancing_mode               = null
│
 
 227:           capacity_scaler              = null
│
 
 228:           description                  = null
│
 
 229:           max_connections              = null
│
 
 230:           max_connections_per_instance = null
│
 
 231:           max_connections_per_endpoint = null
│
 
 232:           max_rate                     = null
│
 
 233:           max_rate_per_instance        = null
│
 
 234:           max_rate_per_endpoint        = null
│
 
 235:           max_utilization              = null
│
 
 236:         },
│
 
 237:       ]
│
 
 238:       iap_config = {
│
 
 239:         enable               = false
│
 
 240:         oauth2_client_id     = ""
│
 
 241:         oauth2_client_secret = ""
│
 
 242:       }
│
 
 243:     }
│
 
 244:     mig2 = {
│
 
 245:       description                     = null
│
 
 246:       protocol                        = "HTTP"
│
 
 247:       port                            = 80
│
 
 248:       port_name                       = "http"
│
 
 249:       timeout_sec                     = 10
│
 
 250:       connection_draining_timeout_sec = null
│
 
 251:       enable_cdn                      = false
│
 
 252:       security_policy                 = null
│
 
 253:       session_affinity                = null
│
 
 254:       affinity_cookie_ttl_sec         = null
│
 
 255:       custom_request_headers          = null
│
 
 256:       custom_response_headers         = null
│
 
 257:       health_check = local.health_check
│
 
 258:       log_config = {
│
 
 259:         enable      = true
│
 
 260:         sample_rate = 1.0
│
 
 261:       }
│
 
 262:       groups = [
│
 
 263:         {
│
 
 264:           group                        = module.mig2.instance_group
│
 
 265:           balancing_mode               = null
│
 
 266:           capacity_scaler              = null
│
 
 267:           description                  = null
│
 
 268:           max_connections              = null
│
 
 269:           max_connections_per_instance = null
│
 
 270:           max_connections_per_endpoint = null
│
 
 271:           max_rate                     = null
│
 
 272:           max_rate_per_instance        = null
│
 
 273:           max_rate_per_endpoint        = null
│
 
 274:           max_utilization              = null
│
 
 275:         },
│
 
 276:       ]
│
 
 277:       iap_config = {
│
 
 278:         enable               = false
│
 
 279:         oauth2_client_id     = ""
│
 
 280:         oauth2_client_secret = ""
│
 
 281:       }
│
 
 282:     }
│
 
 283:     mig3 = {
│
 
 284:       description                     = null
│
 
 285:       protocol                        = "HTTP"
│
 
 286:       port                            = 80
│
 
 287:       port_name                       = "http"
│
 
 288:       timeout_sec                     = 10
│
 
 289:       connection_draining_timeout_sec = null
│
 
 290:       enable_cdn                      = false
│
 
 291:       security_policy                 = null
│
 
 292:       session_affinity                = null
│
 
 293:       affinity_cookie_ttl_sec         = null
│
 
 294:       custom_request_headers          = null
│
 
 295:       custom_response_headers         = null
│
 
 296:       health_check = local.health_check
│
 
 297:       log_config = {
│
 
 298:         enable      = true
│
 
 299:         sample_rate = 1.0
│
 
 300:       }
│
 
 301:       groups = [
│
 
 302:         {
│
 
 303:           group                        = module.mig3.instance_group
│
 
 304:           balancing_mode               = null
│
 
 305:           capacity_scaler              = null
│
 
 306:           description                  = null
│
 
 307:           max_connections              = null
│
 
 308:           max_connections_per_instance = null
│
 
 309:           max_connections_per_endpoint = null
│
 
 310:           max_rate                     = null
│
 
 311:           max_rate_per_instance        = null
│
 
 312:           max_rate_per_endpoint        = null
│
 
 313:           max_utilization              = null
│
 
 314:         },
│
 
 315:       ]
│
 
 316:       iap_config = {
│
 
 317:         enable               = false
│
 
 318:         oauth2_client_id     = ""
│
 
 319:         oauth2_client_secret = ""
│
 
 320:       }
│
 
 321:     }
│
 
 322:   }
│
 
│
 
The given value is not suitable for child module variable "backends"
│
 
defined at ../../variables.tf:81,1-20: element "mig2": attribute
│
 
"cdn_config" is required.
╵
terraform_validate ./examples/shared-vpc 
Success!
 The configuration is valid.
terraform_validate ./modules/dynamic_backends 
╷
│
 
Error: 
Unsupported argument
│
 
│
 
  on main.tf line 209, in resource "google_compute_backend_service" "default":
│
 
 209:       
deafult_ttl
 = lookup(lookup(each.value, "cdn_config", {}), "default_ttl", null)
│
 
│
 
An argument named "deafult_ttl" is not expected here. Did you mean
│
 
"default_ttl"?
╵
terraform_validate ./modules/serverless_negs 
╷
│
 
Error: 
Unsupported argument
│
 
│
 
  on main.tf line 194, in resource "google_compute_backend_service" "default":
│
 
 194:       
deafult_ttl
 = lookup(lookup(each.value, "cdn_config", {}), "default_ttl", null)
│
 
│
 
An argument named "deafult_ttl" is not expected here. Did you mean
│
 
"default_ttl"?
╵
terraform_validate ./test/fixtures/cloudrun 
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/https-redirect 
╷
│
 
Warning: 
Argument is deprecated
│
 
│
 
  with module.example.tls_self_signed_cert.example,
│
 
  on ../../../examples/https-redirect/tls.tf line 23, in resource "tls_self_signed_cert" "example":
│
 
  23:   key_algorithm   = 
tls_private_key.example.algorithm
│
 
│
 
This is now ignored, as the key algorithm is inferred from the
│
 
`private_key_pem`.
╵
Success!
 The configuration is valid, but there were some
validation warnings as shown above.
terraform_validate ./test/fixtures/mig_nat 
╷
│
 
Error: 
Invalid value for module argument
│
 
│
 
  on ../../../examples/mig-nat-http-lb/main.tf line 101, in module "gce-lb-http":
│
 
 101:   backends = 
{
│
 
 102:     default = {
│
 
 103:       description                     = null
│
 
 104:       protocol                        = "HTTP"
│
 
 105:       port                            = 80
│
 
 106:       port_name                       = "http"
│
 
 107:       timeout_sec                     = 10
│
 
 108:       connection_draining_timeout_sec = null
│
 
 109:       enable_cdn                      = false
│
 
 110:       security_policy                 = null
│
 
 111:       session_affinity                = null
│
 
 112:       affinity_cookie_ttl_sec         = null
│
 
 113:       custom_request_headers          = null
│
 
 114:       custom_response_headers         = null
│
 
 115:       health_check = {
│
 
 116:         check_interval_sec  = null
│
 
 117:         timeout_sec         = null
│
 
 118:         healthy_threshold   = null
│
 
 119:         unhealthy_threshold = null
│
 
 120:         request_path        = "/"
│
 
 121:         port                = 80
│
 
 122:         host                = null
│
 
 123:         logging             = null
│
 
 124:       }
│
 
 125:       log_config = {
│
 
 126:         enable      = false
│
 
 127:         sample_rate = null
│
 
 128:       }
│
 
 129:       groups = [
│
 
 130:         {
│
 
 131:           group                        = module.mig.instance_group
│
 
 132:           balancing_mode               = null
│
 
 133:           capacity_scaler              = null
│
 
 134:           description                  = null
│
 
 135:           max_connections              = null
│
 
 136:           max_connections_per_instance = null
│
 
 137:           max_connections_per_endpoint = null
│
 
 138:           max_rate                     = null
│
 
 139:           max_rate_per_instance        = null
│
 
 140:           max_rate_per_endpoint        = null
│
 
 141:           max_utilization              = null
│
 
 142:         }
│
 
 143:       ]
│
 
 144:       iap_config = {
│
 
 145:         enable               = false
│
 
 146:         oauth2_client_id     = ""
│
 
 147:         oauth2_client_secret = ""
│
 
 148:       }
│
 
 149:     }
│
 
 150:   }
│
 
│
 
The given value is not suitable for child module variable "backends"
│
 
defined at ../../../variables.tf:81,1-20: element "default": attribute
│
 
"cdn_config" is required.
╵
terraform_validate ./test/fixtures/multi_certs 
╷
│
 
Warning: 
Argument is deprecated
│
 
│
 
  with module.example.tls_self_signed_cert.example,
│
 
  on ../../../examples/multiple-certs/tls.tf line 25, in resource "tls_self_signed_cert" "example":
│
 
  25:   key_algorithm   = 
tls_private_key.example[count.index].algorithm
│
 
│
 
This is now ignored, as the key algorithm is inferred from the
│
 
`private_key_pem`.
╵
╷
│
 
Error: 
Invalid value for module argument
│
 
│
 
  on ../../../examples/multiple-certs/main.tf line 139, in module "gce-lb-https":
│
 
 139:   backends = 
{
│
 
 140:     default = {
│
 
 141:       description                     = null
│
 
 142:       protocol                        = "HTTP"
│
 
 143:       port                            = 80
│
 
 144:       port_name                       = "http"
│
 
 145:       timeout_sec                     = 10
│
 
 146:       connection_draining_timeout_sec = null
│
 
 147:       enable_cdn                      = false
│
 
 148:       security_policy                 = null
│
 
 149:       session_affinity                = null
│
 
 150:       affinity_cookie_ttl_sec         = null
│
 
 151:       custom_request_headers          = null
│
 
 152:       custom_response_headers         = null
│
 
 153:       health_check = local.health_check
│
 
 154:       log_config = {
│
 
 155:         enable      = true
│
 
 156:         sample_rate = 1.0
│
 
 157:       }
│
 
 158:       groups = [
│
 
 159:         {
│
 
 160:           group                        = module.mig1.instance_group
│
 
 161:           balancing_mode               = null
│
 
 162:           capacity_scaler              = null
│
 
 163:           description                  = null
│
 
 164:           max_connections              = null
│
 
 165:           max_connections_per_instance = null
│
 
 166:           max_connections_per_endpoint = null
│
 
 167:           max_rate                     = null
│
 
 168:           max_rate_per_instance        = null
│
 
 169:           max_rate_per_endpoint        = null
│
 
 170:           max_utilization              = null
│
 
 171:         },
│
 
 172:         {
│
 
 173:           group                        = module.mig2.instance_group
│
 
 174:           balancing_mode               = null
│
 
 175:           capacity_scaler              = null
│
 
 176:           description                  = null
│
 
 177:           max_connections              = null
│
 
 178:           max_connections_per_instance = null
│
 
 179:           max_connections_per_endpoint = null
│
 
 180:           max_rate                     = null
│
 
 181:           max_rate_per_instance        = null
│
 
 182:           max_rate_per_endpoint        = null
│
 
 183:           max_utilization              = null
│
 
 184:         },
│
 
 185:         {
│
 
 186:           group                        = module.mig3.instance_group
│
 
 187:           balancing_mode               = null
│
 
 188:           capacity_scaler              = null
│
 
 189:           description                  = null
│
 
 190:           max_connections              = null
│
 
 191:           max_connections_per_instance = null
│
 
 192:           max_connections_per_endpoint = null
│
 
 193:           max_rate                     = null
│
 
 194:           max_rate_per_instance        = null
│
 
 195:           max_rate_per_endpoint        = null
│
 
 196:           max_utilization              = null
│
 
 197:         },
│
 
 198:       ]
│
 
 199:       iap_config = {
│
 
 200:         enable               = false
│
 
 201:         oauth2_client_id     = ""
│
 
 202:         oauth2_client_secret = ""
│
 
 203:       }
│
 
 204:     }
│
 
 205:     mig1 = {
│
 
 206:       description                     = null
│
 
 207:       protocol                        = "HTTP"
│
 
 208:       port                            = 80
│
 
 209:       port_name                       = "http"
│
 
 210:       timeout_sec                     = 10
│
 
 211:       connection_draining_timeout_sec = null
│
 
 212:       enable_cdn                      = false
│
 
 213:       security_policy                 = null
│
 
 214:       session_affinity                = null
│
 
 215:       affinity_cookie_ttl_sec         = null
│
 
 216:       custom_request_headers          = null
│
 
 217:       custom_response_headers         = null
│
 
 218:       health_check = local.health_check
│
 
 219:       log_config = {
│
 
 220:         enable      = true
│
 
 221:         sample_rate = 1.0
│
 
 222:       }
│
 
 223:       groups = [
│
 
 224:         {
│
 
 225:           group                        = module.mig1.instance_group
│
 
 226:           balancing_mode               = null
│
 
 227:           capacity_scaler              = null
│
 
 228:           description                  = null
│
 
 229:           max_connections              = null
│
 
 230:           max_connections_per_instance = null
│
 
 231:           max_connections_per_endpoint = null
│
 
 232:           max_rate                     = null
│
 
 233:           max_rate_per_instance        = null
│
 
 234:           max_rate_per_endpoint        = null
│
 
 235:           max_utilization              = null
│
 
 236:         },
│
 
 237:       ]
│
 
 238:       iap_config = {
│
 
 239:         enable               = false
│
 
 240:         oauth2_client_id     = ""
│
 
 241:         oauth2_client_secret = ""
│
 
 242:       }
│
 
 243:     }
│
 
 244:     mig2 = {
│
 
 245:       description                     = null
│
 
 246:       protocol                        = "HTTP"
│
 
 247:       port                            = 80
│
 
 248:       port_name                       = "http"
│
 
 249:       timeout_sec                     = 10
│
 
 250:       connection_draining_timeout_sec = null
│
 
 251:       enable_cdn                      = false
│
 
 252:       security_policy                 = null
│
 
 253:       session_affinity                = null
│
 
 254:       affinity_cookie_ttl_sec         = null
│
 
 255:       custom_request_headers          = null
│
 
 256:       custom_response_headers         = null
│
 
 257:       health_check = local.health_check
│
 
 258:       log_config = {
│
 
 259:         enable      = true
│
 
 260:         sample_rate = 1.0
│
 
 261:       }
│
 
 262:       groups = [
│
 
 263:         {
│
 
 264:           group                        = module.mig2.instance_group
│
 
 265:           balancing_mode               = null
│
 
 266:           capacity_scaler              = null
│
 
 267:           description                  = null
│
 
 268:           max_connections              = null
│
 
 269:           max_connections_per_instance = null
│
 
 270:           max_connections_per_endpoint = null
│
 
 271:           max_rate                     = null
│
 
 272:           max_rate_per_instance        = null
│
 
 273:           max_rate_per_endpoint        = null
│
 
 274:           max_utilization              = null
│
 
 275:         },
│
 
 276:       ]
│
 
 277:       iap_config = {
│
 
 278:         enable               = false
│
 
 279:         oauth2_client_id     = ""
│
 
 280:         oauth2_client_secret = ""
│
 
 281:       }
│
 
 282:     }
│
 
 283:     mig3 = {
│
 
 284:       description                     = null
│
 
 285:       protocol                        = "HTTP"
│
 
 286:       port                            = 80
│
 
 287:       port_name                       = "http"
│
 
 288:       timeout_sec                     = 10
│
 
 289:       connection_draining_timeout_sec = null
│
 
 290:       enable_cdn                      = false
│
 
 291:       security_policy                 = null
│
 
 292:       session_affinity                = null
│
 
 293:       affinity_cookie_ttl_sec         = null
│
 
 294:       custom_request_headers          = null
│
 
 295:       custom_response_headers         = null
│
 
 296:       health_check = local.health_check
│
 
 297:       log_config = {
│
 
 298:         enable      = true
│
 
 299:         sample_rate = 1.0
│
 
 300:       }
│
 
 301:       groups = [
│
 
 302:         {
│
 
 303:           group                        = module.mig3.instance_group
│
 
 304:           balancing_mode               = null
│
 
 305:           capacity_scaler              = null
│
 
 306:           description                  = null
│
 
 307:           max_connections              = null
│
 
 308:           max_connections_per_instance = null
│
 
 309:           max_connections_per_endpoint = null
│
 
 310:           max_rate                     = null
│
 
 311:           max_rate_per_instance        = null
│
 
 312:           max_rate_per_endpoint        = null
│
 
 313:           max_utilization              = null
│
 
 314:         },
│
 
 315:       ]
│
 
 316:       iap_config = {
│
 
 317:         enable               = false
│
 
 318:         oauth2_client_id     = ""
│
 
 319:         oauth2_client_secret = ""
│
 
 320:       }
│
 
 321:     }
│
 
 322:   }
│
 
│
 
The given value is not suitable for child module variable "backends"
│
 
defined at ../../../variables.tf:81,1-20: element "mig1": attribute
│
 
"cdn_config" is required.
╵
terraform_validate ./test/fixtures/multi_mig 
Success!
 The configuration is valid.
terraform_validate ./test/setup 
Success!
 The configuration is valid.
 ```

File: autogen/variables.tf.tmpl [hunk 1]
```diff
@@ -118,6 +118,21 @@ variable "backends" {
     })
     {% endif %}
 
+    cdn_config = object({
+      default_ttl = number
+      max_ttl     = number
+      client_ttl  = number
+      cache_mode  = string
```
Comment (villesau) [inline autogen/variables.tf.tmpl:125]: at least following options are possible:



```
    cache_mode                   = "CACHE_ALL_STATIC"
    client_ttl                   = 60
    default_ttl                  = 60
    max_ttl                      = 86400
    negative_caching             = false
    serve_while_stale            = 0
    signed_url_cache_max_age_sec = 0
   ```

Comment (villesau) [inline autogen/variables.tf.tmpl:125]: at least following options are possible:



```
    cache_mode                   = "CACHE_ALL_STATIC"
    client_ttl                   = 60
    default_ttl                  = 60
    max_ttl                      = 86400
    negative_caching             = false
    serve_while_stale            = 0
    signed_url_cache_max_age_sec = 0
   ```

**Comment by benhinchley** [comment]
@bharathkkb Any chance you could give this PR a look? _sorry if you are not the right person, just saw you reviewing another PR_

**Review by bharathkkb — COMMENTED** [review]
Thanks for the PR @benhinchley  and sorry for the delay.
Can we also add an upgrade guide in https://github.com/terraform-google-modules/terraform-google-lb-http/tree/master/docs and enable this in an example so we can verify via a test in https://github.com/terraform-google-modules/terraform-google-lb-http/tree/master/test/integration

**Comment by villers** [comment]
reup this pr ?

**Pull request closed without merging**